File: langs/csharp/src/VMx/Capabilities/CurrentCrud.cs
namespace VMx.Capabilities;

// Capability interfaces from spec/14-capabilities.md §Container-current.

/// <summary>Capability: the implementer can delete its current selection.</summary>
public interface ICurrentDeletable
{
    /// <summary>Returns true when <see cref="DeleteCurrent"/> is valid to call.</summary>
    bool CanDeleteCurrent();

    /// <summary>Deletes the current selection.</summary>
    void DeleteCurrent();
}

/// <summary>Capability: the implementer can update its current selection.</summary>
public interface ICurrentUpdatable
{
    /// <summary>Returns true when <see cref="UpdateCurrent"/> is valid to call.</summary>
    bool CanUpdateCurrent();

    /// <summary>Updates the current selection.</summary>
    void UpdateCurrent();
}

File: langs/python/src/vmx/capabilities/current_crud.py
"""Container-current capability contracts. See spec/14-capabilities.md §Container-current."""

from __future__ import annotations

from abc import ABC, abstractmethod


class ICurrentDeletable(ABC):
    @abstractmethod
    def can_delete_current(self) -> bool: ...

    @abstractmethod
    def delete_current(self) -> None: ...


class ICurrentUpdatable(ABC):
    @abstractmethod
    def can_update_current(self) -> bool: ...

    @abstractmethod
    def update_current(self) -> None: ...
